Docs/api composable use route (#4776)
nuxt/framework: docs/content/3.api/1.composables/use-route.md

# `useRoute`

The `useRoute` composable returns the current route and must be called in a `setup` function, plugin, or route middleware.

Within the template of a Vue component, you can access the route using `$route`.

## Example

In the following example, we call an API via `useAsyncData` using a dynamic page parameter - `slug` - as part of the URL.

```html [~/pages/[slug].vue]
<template>
    <h1>{{ data.title }}</h1>
    <p>{{ data.description }}</p>
</template>
<script setup>
const route = useRoute();
const slug = route.params.slug;
const { data, pending, error, refresh } = await useAsyncData("mountain", () =>
    $fetch(`https://api.nuxtjs.dev/mountains/${slug}`)
);
</script>
```

If you need to access the route query parameters (for example `example` in the path `/test?example=true`), then you can use `route.query` instead of `route.params`.

Apart from dynamic parameters and query parameters, `useRoute` also provides the following computed references related to the current route:

> * **fullPath**: encoded URL associated to the current route that contains path, query and hash

Reviewer: ```suggestion
* **fullPath**: encoded URL associated with the current route that contains path, query and hash
```

# `useRoute`

> `useRoute` composable returns the location of current route and must be called inside `setup()` function.

Reviewer 2: ```suggestion
The `useRoute` composable returns the current route and must be called in a `setup` function, plugin, or route middleware.
```

> `useRoute` composable is accessed in `setup()` function using `useRoute()`, and directly inside the template of Vue component using `$route`.

Reviewer 2: ```suggestion
Within the template of a Vue component, you can access the route using `$route`.
```

> 
> We can expose possible dynamic parameters of the route via `useRoute` composable. 

Reviewer 2: ```suggestion
```

## Example

In the following example, we call an API via `useAsyncData` using a dynamic page parameter - `slug` - as part of the URL.

```html [~/pages/[slug].vue]
<template>
    <h1>{{ data.title }}</h1>
    <p>{{ data.description }}</p>
</template>
<script setup>
const route = useRoute();
const slug = route.params.slug;
const { data, pending, error, refresh } = await useAsyncData("mountain", () =>
    $fetch(`https://api.nuxtjs.dev/mountains/${slug}`)
);
</script>
```

> If API endpoint requires query parameters to construct URL, then you can use `route.query` instead of `route.params`.

Reviewer 2: ```suggestion
If you need to access the route query parameters (for example `example` in the path `/test?example=true`), then you can use `route.query` instead of `route.params`.
```
